app/segments/(team)/chooseTeam.js

```diff
@@ -99,22 +104,56 @@ function ChooseTeam() {
-                  name: currentUserInfo["displayName"],
-                  // hardcoded pfp string for now, add pfp upload to profile settings in future PR
-                  pfp: "",
-                  // hardcoded "player" role for now, add role selection to profile settings in future PR
-                  role: "player",
-                  uid: currentUserId,
-                  assigned_data: [],
-                  uniqueDrills: [],
-                });
-                setCurrentUserId(currentUserId);
-                await invalidateMultipleKeys(queryClient, [
-                  ["userInfo", { userId: currentUserId }],
-                ]);
-                router.replace("/");
+                let completed = false;
+
+                const unregisterAuthObserver = onIdTokenChanged(
+                  auth,
+                  async (user) => {
+                    if (user && !completed) {
+                      await user.reload();
+                      if (user.emailVerified) {
+                        completed = true;
+                        console.log(
+                          "Email successfully verified. Adding user to team, then redirecting to Home Screen",
+                        );
+                        showSnackBar(
+                          "Email successfully verified. Adding user to team, then redirecting to Home Screen",
+                        );
+
+                        // Update Firestore document
+                        await setDoc(
+                          doc(db, "teams", "1", "users", currentUserId),
+                          {
+                            name: currentUserInfo["displayName"],
+                            pfp: "",
+                            role: "player",
+                            uid: currentUserId,
+                            assigned_data: [],
+                            uniqueDrills: [],
+                          },
+                        );
+                        setCurrentUserId(currentUserId);
+                        await invalidateMultipleKeys(queryClient, [
+                          ["userInfo", { userId: currentUserId }],
+                        ]);
+                        unregisterAuthObserver();
+                        // Navigate to the next page
+                        router.replace("/");
+                      } else {
+                        console.log("Error: Email Not Verified Yet, Try Again");
+                        showDialog(
+                          "Error",
+                          "Email Not Verified Yet, Try Again",
+                        );
+                        // Allow re-attempt
+                        completed = false;
+                      }
+                    }
+                  },
+                );
+
+                // Trigger a user reload to update the token and invoke the listener
+                await auth.currentUser.reload();
```

Comment:
I like your placement of the reload here. Should we also place one in a pull to refresh for this page?

Reply:
duplicate comment?